TESTING.asciidoc

```diff
@@ -325,26 +326,37 @@ vagrant plugin install vagrant-cachier
 
 . Download and smoke test the VMs with `./gradlew vagrantSmokeTest` or
 `./gradlew -Pvagrant.boxes=all vagrantSmokeTest`. The first time you run this it will
-download the base images and provision the boxes and immediately quit. If you
-you this again it'll skip the download step.
+download the base images and provision the boxes and immediately quit. Downloading all
+the images may take a long time. After the images are already on your machine, they won't
+be downloaded again unless they have been updated to a new version.
 
 . Run the tests with `./gradlew packagingTest`. This will cause Gradle to build
 the tar, zip, and deb packages and all the plugins. It will then run the tests
 on ubuntu-1404 and centos-7. We chose those two distributions as the default
 because they cover deb and rpm packaging and SyvVinit and systemd.
 
-You can run on all the VMs by running `./gradlew -Pvagrant.boxes=all
-packagingTest`. You can run a particular VM with a command like `./gradlew
--Pvagrant.boxes=oel-7 packagingTest`. See `./gradlew tasks` for a complete list
-of available vagrant boxes for testing. It's important to know that if you
-interrupt any of these Gradle commands then the boxes will remain running and
-you'll have to terminate them with `./gradlew stop`.
+You can choose which boxes to test by setting the `-Pvagrant.boxes` project property. For
+example, to test all boxes, run `./gradlew -Pvagrant.boxes=all`. For a complete list of
+available boxes, run `./gradlew :qa:vagrant:tasks --all`. All of the following values are
+supported by the `vagrant.boxes` property
+
+* `sample` - The default, only chooses ubuntu-1404 and centos-7
+* `all` - All configured boxes, i.e. all linux boxes and any configured Windows
+boxes.
+* `linux-all` - All linux boxes.
+* `windows-all` - All configured Windows boxes. If there are none configured when this
```

Comment:
Thank you for avoiding leniency!

Reply:
Out of curiosity, do we fail if an empty string is passed to `vagrant.boxes`? (separate issue if we don't)

Reply:
It does fail with an empty string, because it ends up looking for a box with the name of empty string and can't find it